app/src/main/java/com/rootstrap/android/network/providers/ServiceProviderModule.kt

```diff
@@ -8,16 +8,16 @@ import com.squareup.moshi.Moshi
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
-import dagger.hilt.android.components.ApplicationComponent
 import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import dagger.hilt.components.SingletonComponent
 import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
 import retrofit2.Retrofit
 import retrofit2.converter.moshi.MoshiConverterFactory
 import javax.inject.Singleton
 
 @Module
-@InstallIn(ApplicationComponent::class)
+@InstallIn(SingletonComponent::class)
```

Comment:
Same as above

Reply:
ApplicationComponent was renamed to SingletonComponent to allow the usage of Hilt in non-Android modules. You can check this release notes for further information : https://github.com/google/dagger/releases/tag/dagger-2.28.2